airflow/models/taskinstance.py

```diff
@@ -1702,7 +1702,8 @@ def handle_failure(
             session.add(Log(State.FAILED, self))
 
             # Log failure duration
-            session.add(TaskFail(task, self.execution_date, self.start_date, self.end_date))
+            dag_run = self.get_dagrun()  # self.dag_run not populated by refresh_from_db
```

Comment:
I believe this is correct:
```suggestion
            dag_run = self.get_dagrun(run_id=self.run_id)  # self.dag_run not populated by refresh_from_db
```

Reply:
Otherwise you will get first (random) from the list of all the dagruns.

Reply:
Ah no. sorry. Wrong `get_dagrun` . All ok.

Reply:
`self.get_dagrun()` already considers `self.run_id` so I don't think this is needed.

Reply:
But

```suggestion
            dag_run = self.get_dagrun(session=session)  # self.dag_run not populated by refresh_from_db
```

Reply:
yeah - looked at wrong get_dagrun() (From `dag.py`)

Reply:
Right - session should be added.